plugins/speccy.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+from cloudbot import hook
+
+import re
+import requests
+
+from bs4 import BeautifulSoup as BS
+from contextlib import closing
+
+url_re = re.compile('https?:\/\/speccy.piriform.com\/results\/[A-z0-9]+', re.I)
+
+
+@hook.regex(url_re)
+def get_speccy_url(message, match, chan, nick):
+
+    with closing(requests.get(match.group(), stream=True,
+                              timeout=3)) as response:
```

Comment:
As you don't read the body as a stream, specifying `stream=True` has no effect, The timeout is also quite low, no reason for that to be honest, as even if the request is slow, other things on the bot will continue to run fine. Waiting on a response from a server is very light on system resources,